### DISCUSSION TITLE
Memory game tutorial, 'Polishing the Tile': Where is the value of `open_curtain` coming from?

### DISCUSSION THREAD

**Discussion by runiq** [slint-ui/slint#2966] [Q&A] [answered]
I'm going through the tutorial and I kind of got stuck on the [Polishing the Tile](https://slint.dev/releases/1.0.2/docs/tutorial/rust/polishing_the_tile) chapter. In it, there's the Boolean `open_curtain` property which appears to default to `false`, but nowhere in the code is it explicitly set to `false`. I've tried looking through the Slint docs for 'default values' or 'default property values', to no avail. Can someone explain where the default value of `false` for `open_curtain` is coming from?

**Comment by ogoffart** [comment, marked as answer]
When a property is not set, it is initialized to the default value of the type, which for book, is false.

It's true that the docs don't say that, we should fix that.

**Reply by runiq** [reply]
Thanks, that clears it up. :) Is [this](https://github.com/slint-ui/slint/blob/4820d29bcb3d1460ac5fb3df012ebf281ce16c76/internal/core/rtti.rs#L17-L25) the place where `Default` is derived for the Boolean value type in Slint?

**Reply by ogoffart** [reply]
Well, not really.  `bool` implements Default in rust, and the property of type bool will ceate a Property<bool> whose default value will be the one of the Default trait.

There are other places where we determine what's the default value, for example in https://github.com/slint-ui/slint/blob/718e4df6dee35e1376d802589b7a978bcd65bc24/internal/compiler/expression_tree.rs#L1211

**Reply by runiq** [reply]
Ah, thanks! Looks like it's this one: https://github.com/slint-ui/slint/blob/718e4df6dee35e1376d802589b7a978bcd65bc24/internal/compiler/expression_tree.rs#L1237

**Comment by ogoffart** [comment]
I added some docs in https://github.com/slint-ui/slint/pull/2993
